### ISSUE TITLE
Application deployment failure

### ISSUE THREAD

**Issue by json-lbtn** [Qovery/terraform-provider-qovery#45]
Hello there 🖖 

There is apparently no way to deploy multiple application simultaneously. Actually, we need to set up `depends_on` attributes in order to serialize the whole deployment.

Also, deploying applications in a "STOPPED" state raises an exception about not being able to stop an application which is aleady stopped.  It is automatically trying to stop it even if it's deployed as stopped isn't it ?

Thank you for your reading : )

**Comment by benamib** [comment]
Hello @json-lbtn, thank you for your feedbacks 👍 

Regarding the deployment of multiple applications; with the actual design of the provider with the `state` field in the application itself it is not possible to deploy everything at the same time. 
This is because terraform creates every app sequentially and the deployment is handle at the creation step. 

To fix this and allow multiple app to be deployed at the same time, the only way I can think of would be to add a new resource `qovery_deployment` to which you would give the ids of apps you want to deploy.

Something like this:
```terraform
qovery_deployment {
  RUNNING = [
      qovery_application.my_app_1.id,
      qovery_application.my_app_2.id
  ]
}
```

I'm open to any suggestion on this one, let me know how you find this solution. 

Also, I'll be taking a look at the issue when deploying a `STOPPED` application. 
Like you said, the application is created in a stopped state named `READY`, since it's different than`STOPPED` it still tries to stop it.

**Comment by json-lbtn** [comment]
Hello @benamib 🍃 
I hope you are doing well !

The `qovery_deployment` resource solution would be clearly appreciated. It would provide a great experience to developers if we also benefit of https://github.com/Qovery/terraform-provider-qovery/issues/16 in the next versions. 
Can't wait to try it this way !

I guess we wouldn't make a use of the `state` application/database attribute anymore with this new resource. 
Like, every resource will naturally be created in a `STOPPED` state.
In this case, I think that doing it this way with `RUNNING = [` isn't that necessary.
I'm just saying, but what about deployment modes instead of deployment states as we will always give `RUNNING` wanted state apps. Something like this :
```tf
qovery_deployment {
  bulk = [
      qovery_application.my_app_1.id,
      qovery_application.my_app_2.id
  ]
}
```
> Which allows to deploy a bulk of applications, no matter the order.
And something like that :
```tf
qovery_deployment {
  tree = {
    "${qovery_application.my_app_1.id}" = [
      qovery_application.my_app_2.id,
      qovery_application.my_app_3.id
    ],
    "${qovery_application.my_app_4.id}" = [
      qovery_application.my_app_5.id
    ],
  }
}
```
> For an ordered deployment (which takes care of dependencies).
I'm just saying. I don't know, maybe we also can use the `depends_on` resources properties to build the deployment order.


And thanks for the review about the `STOPPED` deployment state. At the moment it's mostly a detail. It was just about giving a little feedback.

**Comment by mzottolaqovery** [comment]
Hello @json-lbtn,
We are currently working on integrating the deployment stages in terraform. This will allow to deploy services in a predefined order.
I'll let you know here when this is done

**Comment by mzottolaqovery** [comment]
Hello @json-lbtn ,
The deployment feature is now available from [v0.15.0](https://github.com/Qovery/terraform-provider-qovery/releases/tag/v0.15.0), you can now create deployment stages, order them, and deploy the whole environment
